pkg/kwok/server/debugging_attach.go

```diff
@@ -21,21 +21,33 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/emicklei/go-restful/v3"
 	"k8s.io/apimachinery/pkg/types"
 	clientremotecommand "k8s.io/client-go/tools/remotecommand"
 
+	"sigs.k8s.io/kwok/pkg/apis/internalversion"
 	"sigs.k8s.io/kwok/pkg/kwok/server/remotecommand"
+	"sigs.k8s.io/kwok/pkg/log"
+	"sigs.k8s.io/kwok/pkg/utils/slices"
 )
 
 // AttachContainer attaches to a container in a pod,
 // copying data between in/out/err and the container's stdin/stdout/stderr.
-func (s *Server) AttachContainer(ctx context.Context, podName, podNamespace string, uid types.UID, container string, in io.Reader, out, err io.WriteCloser, tty bool, resize <-chan clientremotecommand.TerminalSize) error {
-	// TODO: Configure and implement the attach streamer
-	msg := fmt.Sprintf("TODO: AttachContainer(%q, %q)", podNamespace+"/"+podName, container)
-	_, _ = out.Write([]byte(msg))
-	return nil
+func (s *Server) AttachContainer(ctx context.Context, podName, podNamespace string, uid types.UID, containerName string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool, resize <-chan clientremotecommand.TerminalSize) error {
+	attach, err := s.getPodAttach(podName, podNamespace, containerName)
+	if err != nil {
+		return err
+	}
+	opts := &logOptions{
+		tail:      -1, // -1 by default which means read all logs.
```

Comment:
I think the value should be 0. I tried to attach and it won't output the history.

Reply:
I based it on the timestamp, using since as `time.Now()` to stream everything starting at the current time. 
It seemed more appropriate to stream starting at the current time but I'm not sure. I'll move it to tail and stream from that point :relaxed: That is why I used timestamp in the future in tests :relaxed: